DragonSlayer.cpp

```diff
@@ -1,16 +1,28 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
+#include "AttackItem.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string charName, int hp, int armor) : Character(hp, armor, 4), name(charName)
+{
+    helpfulItems = makeHelpfulItems(1);
+    defensiveItems = makeDefensiveItems(2);
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        if (attackItem != nullptr && dragon->getHP() > 0)
+            attackItem->use(this);  
+            attackItem.reset(); // Single use item
+        
```

Comment:
The lack of curly braces here means that attack item is being reset always. 
This is not Python, where indentation determines scope. 
Curly braces determine scope.
- [ ] revise